spec/builtin-history.test.sh

```diff
@@ -32,29 +34,65 @@ history -a
 ^D
 ## END
 
-#### history -r
-rm -f tmp
-echo 'foo' > tmp
 
-echo '
+#### history -w
```

Comment:
I guess this one would be "history -w writes history from memory to disk" ? 

or the spec test seems to be doing something slightly different

It could use some comments in the body too